helper/validation/write_only_test.go

```diff
@@ -0,0 +1,780 @@
+						}),
+					}),
+				}),
+			},
+			expectedDiags: diag.Diagnostics{
+				{
+					Severity: diag.Warning,
+					Summary:  "Available Write-Only Attribute Alternative",
+					Detail: "The attribute oldAttribute has a WriteOnly version writeOnlyAttribute available. " +
+						"Use the WriteOnly version of the attribute when possible.",
+					AttributePath: cty.Path{
+						cty.GetAttrStep{Name: "set_nested_block"},
+						cty.IndexStep{Key: cty.ObjectVal(map[string]cty.Value{
+							"oldAttribute": cty.StringVal("value"),
+						})},
+						cty.GetAttrStep{Name: "oldAttribute"},
+					},
+				},
+			},
+		},
+		"set nested block: multiple oldAttribute set returns multiple warning diags": {
+			oldAttributePath: cty.Path{
+				cty.GetAttrStep{Name: "set_nested_block"},
+				cty.IndexStep{Key: cty.UnknownVal(cty.Object(nil))},
+				cty.GetAttrStep{Name: "oldAttribute"},
+			},
+			validateConfigReq: schema.ValidateResourceConfigFuncRequest{
+				WriteOnlyAttributesAllowed: true,
+				RawConfig: cty.ObjectVal(map[string]cty.Value{
+					"id": cty.NullVal(cty.String),
+					"set_nested_block": cty.SetVal([]cty.Value{
+						cty.ObjectVal(map[string]cty.Value{
+							"oldAttribute":       cty.StringVal("value1"),
+							"writeOnlyAttribute": cty.NullVal(cty.String),
+						}),
+						cty.ObjectVal(map[string]cty.Value{
+							"oldAttribute":       cty.NullVal(cty.String),
+							"writeOnlyAttribute": cty.StringVal("value2"),
+						}),
+						cty.ObjectVal(map[string]cty.Value{
+							"oldAttribute":       cty.StringVal("value3"),
+							"writeOnlyAttribute": cty.NullVal(cty.String),
+						}),
+					}),
+				}),
+			},
+			expectedDiags: diag.Diagnostics{
+				{
+					Severity: diag.Warning,
+					Summary:  "Available Write-Only Attribute Alternative",
+					Detail: "The attribute oldAttribute has a WriteOnly version writeOnlyAttribute available. " +
+						"Use the WriteOnly version of the attribute when possible.",
+					AttributePath: cty.Path{
+						cty.GetAttrStep{Name: "set_nested_block"},
+						cty.IndexStep{Key: cty.ObjectVal(map[string]cty.Value{
+							"oldAttribute":       cty.StringVal("value1"),
+							"writeOnlyAttribute": cty.NullVal(cty.String),
+						})},
+						cty.GetAttrStep{Name: "oldAttribute"},
+					},
+				},
+				{
+					Severity: diag.Warning,
+					Summary:  "Available Write-Only Attribute Alternative",
+					Detail: "The attribute oldAttribute has a WriteOnly version writeOnlyAttribute available. " +
+						"Use the WriteOnly version of the attribute when possible.",
+					AttributePath: cty.Path{
+						cty.GetAttrStep{Name: "set_nested_block"},
+						cty.IndexStep{Key: cty.ObjectVal(map[string]cty.Value{
+							"oldAttribute":       cty.StringVal("value3"),
+							"writeOnlyAttribute": cty.NullVal(cty.String),
+						})},
+						cty.GetAttrStep{Name: "oldAttribute"},
+					},
+				},
+			},
+		},
+		"map nested block: oldAttribute and writeOnlyAttribute map returns warning diags": {
```

Comment:
Similar question to one I had above, are map nested blocks in SDKv2? Or is this a map attribute equivalent